src/node.rs

```diff
@@ -1,37 +1,40 @@
 #[derive(Clone, PartialEq, Debug)]
 pub enum Node {
-    Atom(Atom),
-    List(List),
-    Type(i64)
+    Atom(&'static str),
+    Pair(Box<Node>, Box<Node>),
+    Type(Type),
 }
 
-pub type Atom = &'static str;
-
-// =====================================
-// List
-// =====================================
-pub type List = Vec<Node>;
+#[derive(Clone, PartialEq, Debug)]
+pub enum Type {
+    Universe(i64),
+    Atom,
+    Pair(Box<Type>, Box<Type>)
+}
 
-pub fn cons(list: List, atom: Atom) -> Node {
-    let mut cloned_list = list.clone();
-    cloned_list.push(Node::Atom(atom));
-    Node::List(cloned_list)
+pub fn cons(node: Node, atom: Node) -> Node {
+    Node::Pair(Box::new(node), Box::new(atom))
 }
 
-pub fn car(list: List) -> Node {
-    list.first().unwrap().clone()
+pub fn car(node: Node) -> Result<Node, &'static str> {
```

Comment:
```suggestion
impl Node {
    pub fn car(self) -> Result<Self, &'static str> {
        match node {
            Node::Pair(a, _) => Result::Ok(*a),
            _ => Result::Err("Can only apply car to Pair")
        }
    }
}
```

Same goes for `cdr` and `cons` btw.
```rust
pub fn cons(node: Self, atom: Self) -> Self { ... }
pub fn cdr(self) -> Result<Self, &'static str> { ... }
```

Reply:
Oops, this is done in #4